src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js

```diff
@@ -96,15 +100,27 @@ export const NewNanoContractTransactionRequest = ({ ncTxRequest }) => {
   const onAcceptTransaction = () => {
     // Update the caller with the address selected by the user.
     const acceptedNc = { ...nc, caller: ncAddress };
+    // Restore ready status to Nano Contract registration state
+    dispatch(nanoContractRegisterReady());
```

Comment:
I understood that you need to dispatch this from the component, but I think we should do this in the screen unmount

This way we don't have to do this twice

Reply:
Oh! You right! I'll change this.

Reply:
Done: 525d756897d748ea72479d9c09a5dadff794bfad